src/main/java/org/primefaces/model/TreeNodeChildren.java

```diff
@@ -214,4 +216,24 @@ private void updateRowKeys(TreeNode node) {
             }
         }
     }
+
+    private void updateRowKeys(int index, TreeNode node) {
+        int childCount = node.getChildCount();
+        if (childCount > 0) {
+            for (int i = index; i < childCount; ++i) {
+                TreeNode childNode = (TreeNode) node.getChildren().get(i);
+                String childRowKey = node.getParent() == null ? String.valueOf(i) : node.getRowKey() + "_" + i;
```

Comment:
In these 3 updateRowKeys methods, these lines are duplicated, could you please refactor that a bit so it can be reused:
```java
String childRowKey = node.getParent() == null ? String.valueOf(i) : node.getRowKey() + "_" + i;
childNode.setRowKey(childRowKey);
updateRowKeys(childNode);
```